cpp/src/arrow/compute/kernels/aggregate_internal.h

```diff
@@ -52,6 +52,16 @@ struct FindAccumulatorType<I, enable_if_floating_point<I>> {
   using Type = DoubleType;
 };
 
+template <typename I>
+struct FindAccumulatorType<I, enable_if_decimal32<I>> {
```

Comment:
Speaking of which, are there any tests for summing decimal32 and decimal64?

Reply:
Not yet, I was going to cover that when I do the rest of the Kernel additions for decimal32/decimal64 in a follow-up PR.

These were only added here to make everything compile due to adding the new types to the generalized `enable_if_decimal...` template helpers